src/extension.ts

```diff
@@ -222,14 +220,22 @@ async function sourceRosAndWorkspace(): Promise<void> {
         } catch (err) {
             vscode.window.showErrorMessage(`Could not source the setup file for ROS distro "${distro}".`);
         }
-    } else if (typeof process.env.ROS_DISTRO !== "undefined") {
+    } else if (process.env.ROS_DISTRO && process.env.ROS_DISTRO.trim().length) {
```

Comment:
Can it be simplified to this?

```suggestion
    } else if (process.env.ROS_DISTRO) {
```

Reply:
good point!